ouroboros-consensus-diffusion/test/consensus-test/Test/Consensus/BlockTree.hs

```diff
@@ -149,19 +150,17 @@ findPath source target blockTree = do
 --                      ╰─────3──4─────5
 --
 -- Returns a list of strings intended to be catenated with a newline.
-prettyPrint :: AF.HasHeader blk => BlockTree blk -> [String]
+prettyPrint :: forall blk . AF.HasHeader blk => BlockTree blk -> [String]
 prettyPrint blockTree = do
   let honestFragment = btTrunk blockTree
-  let advFragment = btbSuffix $ head $ btBranches blockTree
 
   let (oSlotNo, oBlockNo) = slotAndBlockNoFromAnchor $ AF.anchor honestFragment
   let (hSlotNo, _) = slotAndBlockNoFromAnchor $ AF.headAnchor honestFragment
 
-  let (aoSlotNo, _) = slotAndBlockNoFromAnchor $ AF.anchor advFragment
-  let (ahSlotNo, _) = slotAndBlockNoFromAnchor $ AF.headAnchor advFragment
+      (advBranches, aoSlotNos, ahSlotNos) = unzip3 (branch oSlotNo . btbSuffix <$> btBranches blockTree)
 
-  let firstSlotNo = min oSlotNo aoSlotNo
-  let lastSlotNo = max hSlotNo ahSlotNo
+  let firstSlotNo = minimum (oSlotNo : aoSlotNos)
+  let lastSlotNo = maximum (hSlotNo : ahSlotNos)
 
   -- FIXME: only handles two fragments at this point. not very hard to make it
```

Comment:
This comment is out of date. (Sorry, I can't make a suggestion on those lines as the last one isn't in the diff.

Reply:
yeah I thought I'll leave it because the "in a clean way" part was still true 😅